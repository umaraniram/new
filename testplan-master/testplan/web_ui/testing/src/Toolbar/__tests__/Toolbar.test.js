import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import Toolbar from '../Toolbar';
import {
  TOOLBAR_BUTTONS_BATCH,
  TOOLBAR_BUTTONS_INTERACTIVE
} from "../../Common/defaults";
import {ResetButton} from "../InteractiveButtons";

function defaultProps() {
  return {
    status: 'passed',
    buttons: TOOLBAR_BUTTONS_BATCH,
    handleNavFilter: jest.fn(),
  };
}

describe('Toolbar', () => {
  let props;
  let mountedToolbar;
  const renderToolbar = () => {
    if (!mountedToolbar) {
      mountedToolbar = shallow(
        <Toolbar {...props} />
      );
    }
    return mountedToolbar;
  };

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    mountedToolbar = undefined;
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    props.handleNavFilter.mockClear();
  });

  it('shallow renders without crashing', () => {
    renderToolbar();
  });

  it('shallow renders the correct HTML structure for batch mode', () => {
    const toolbar = renderToolbar();
    expect(toolbar).toMatchSnapshot();
  });

  it('shallow renders the correct HTML structure for interactive mode', () => {
    props.buttons = TOOLBAR_BUTTONS_INTERACTIVE;
    const toolbar = renderToolbar();
    expect(toolbar).toMatchSnapshot();
  });

  it('uses the failed style when status is failed', () => {
    props.status = 'failed';
    const toolbar = renderToolbar();
    const container = toolbar.find('Collapse').get(0);
    expect(container.props.className).toMatch(/toolbar.+toolbarFailed/);
  });

  it('uses the neutral style when status is unknown', () => {
    props.status = 'unknown';
    const toolbar = renderToolbar();
    const container = toolbar.find('Collapse').get(0);
    expect(container.props.className).toMatch(/toolbar.+toolbarUnknown/);
  });

  it('inserts extra buttons into the toolbar', () => {
    const resetCbk = jest.fn()
    props.extraButtons = [ResetButton(resetCbk, false)];
    const toolbar = renderToolbar();
    expect(toolbar).toMatchSnapshot();
  });

});
