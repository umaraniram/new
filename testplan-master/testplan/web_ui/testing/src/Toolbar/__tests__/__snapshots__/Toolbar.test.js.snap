// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar inserts extra buttons into the toolbar 1`] = `
<div>
  <Navbar
    className="toolbar_l6n5rs"
    expand="md"
    light={true}
    tag="nav"
  >
    <div
      className="filterBox_1kzbnzj"
    >
      <FilterBox
        handleNavFilter={[MockFunction]}
      />
    </div>
    <Collapse
      appear={false}
      className="toolbar_l6n5rs-o_O-toolbarPassed_1bjpgj2"
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          key="reset-button"
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    640,
                    512,
                    Array [],
                    "f55a",
                    "M576 64H205.26A63.97 63.97 0 0 0 160 82.75L9.37 233.37c-12.5 12.5-12.5 32.76 0 45.25L160 429.25c12 12 28.28 18.75 45.25 18.75H576c35.35 0 64-28.65 64-64V128c0-35.35-28.65-64-64-64zm-84.69 254.06c6.25 6.25 6.25 16.38 0 22.63l-22.62 22.62c-6.25 6.25-16.38 6.25-22.63 0L384 301.25l-62.06 62.06c-6.25 6.25-16.38 6.25-22.63 0l-22.62-22.62c-6.25-6.25-6.25-16.38 0-22.63L338.75 256l-62.06-62.06c-6.25-6.25-6.25-16.38 0-22.63l22.62-22.62c6.25-6.25 16.38-6.25 22.63 0L384 210.75l62.06-62.06c6.25-6.25 16.38-6.25 22.63 0l22.62 22.62c6.25 6.25 6.25 16.38 0 22.63L429.25 256l62.06 62.06z",
                  ],
                  "iconName": "backspace",
                  "prefix": "fas",
                }
              }
              inverse={false}
              key="toolbar-reset"
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Reset state"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="info"
              inverse={false}
              key="toolbar-info"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Info"
              transform={null}
            />
          </div>
        </NavItem>
        <UncontrolledDropdown
          inNavbar={true}
          nav={true}
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <DropdownToggle
              aria-haspopup={true}
              className="toolbar_l6n5rs-o_O-toolbarPassed_1bjpgj2"
              color="secondary"
              nav={true}
            >
              <FontAwesomeIcon
                border={false}
                className="toolbarButton_1d6e8oq"
                fixedWidth={false}
                flip={null}
                icon="filter"
                inverse={false}
                key="toolbar-filter"
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Choose filter"
                transform={null}
              />
            </DropdownToggle>
          </div>
          <DropdownMenu
            className="filterDropdown_biw13d"
            flip={true}
            tag="div"
          >
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={true}
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                  value="all"
                />
                 
                All
              </Label>
            </DropdownItem>
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={false}
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                  value="fail"
                />
                 
                Failed only
              </Label>
            </DropdownItem>
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={false}
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                  value="pass"
                />
                 
                Passed only
              </Label>
            </DropdownItem>
            <DropdownItem
              divider={true}
              tag="button"
              toggle={true}
            />
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={false}
                  name="displayEmptyTest"
                  onChange={[Function]}
                  type="checkbox"
                />
                 
                Hide empty testcase
              </Label>
            </DropdownItem>
          </DropdownMenu>
        </UncontrolledDropdown>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="print"
              inverse={false}
              key="toolbar-print"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Print page"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="tags"
              inverse={false}
              key="toolbar-tags"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Display tags"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="question-circle"
              inverse={false}
              key="toolbar-question"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Help"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <a
            className="buttonsBar_6zsntc"
            href="http://testplan.readthedocs.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="book"
              inverse={false}
              key="toolbar-document"
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Documentation"
              transform={null}
            />
          </a>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="HelpModal"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Help
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      This is filter box help!
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="light"
        onClick={[Function]}
        tag="button"
      >
        Close
      </Button>
    </ModalFooter>
  </Modal>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="infoModal"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    size="lg"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Information
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      No information to display.
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="light"
        onClick={[Function]}
        tag="button"
      >
        Close
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;

exports[`Toolbar shallow renders the correct HTML structure for batch mode 1`] = `
<div>
  <Navbar
    className="toolbar_l6n5rs"
    expand="md"
    light={true}
    tag="nav"
  >
    <div
      className="filterBox_1kzbnzj"
    >
      <FilterBox
        handleNavFilter={[MockFunction]}
      />
    </div>
    <Collapse
      appear={false}
      className="toolbar_l6n5rs-o_O-toolbarPassed_1bjpgj2"
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="info"
              inverse={false}
              key="toolbar-info"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Info"
              transform={null}
            />
          </div>
        </NavItem>
        <UncontrolledDropdown
          inNavbar={true}
          nav={true}
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <DropdownToggle
              aria-haspopup={true}
              className="toolbar_l6n5rs-o_O-toolbarPassed_1bjpgj2"
              color="secondary"
              nav={true}
            >
              <FontAwesomeIcon
                border={false}
                className="toolbarButton_1d6e8oq"
                fixedWidth={false}
                flip={null}
                icon="filter"
                inverse={false}
                key="toolbar-filter"
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Choose filter"
                transform={null}
              />
            </DropdownToggle>
          </div>
          <DropdownMenu
            className="filterDropdown_biw13d"
            flip={true}
            tag="div"
          >
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={true}
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                  value="all"
                />
                 
                All
              </Label>
            </DropdownItem>
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={false}
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                  value="fail"
                />
                 
                Failed only
              </Label>
            </DropdownItem>
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={false}
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                  value="pass"
                />
                 
                Passed only
              </Label>
            </DropdownItem>
            <DropdownItem
              divider={true}
              tag="button"
              toggle={true}
            />
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={false}
                  name="displayEmptyTest"
                  onChange={[Function]}
                  type="checkbox"
                />
                 
                Hide empty testcase
              </Label>
            </DropdownItem>
          </DropdownMenu>
        </UncontrolledDropdown>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="print"
              inverse={false}
              key="toolbar-print"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Print page"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="tags"
              inverse={false}
              key="toolbar-tags"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Display tags"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="question-circle"
              inverse={false}
              key="toolbar-question"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Help"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <a
            className="buttonsBar_6zsntc"
            href="http://testplan.readthedocs.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="book"
              inverse={false}
              key="toolbar-document"
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Documentation"
              transform={null}
            />
          </a>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="HelpModal"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Help
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      This is filter box help!
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="light"
        onClick={[Function]}
        tag="button"
      >
        Close
      </Button>
    </ModalFooter>
  </Modal>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="infoModal"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    size="lg"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Information
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      No information to display.
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="light"
        onClick={[Function]}
        tag="button"
      >
        Close
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;

exports[`Toolbar shallow renders the correct HTML structure for interactive mode 1`] = `
<div>
  <Navbar
    className="toolbar_l6n5rs"
    expand="md"
    light={true}
    tag="nav"
  >
    <div
      className="filterBox_1kzbnzj"
    >
      <FilterBox
        handleNavFilter={[MockFunction]}
      />
    </div>
    <Collapse
      appear={false}
      className="toolbar_l6n5rs-o_O-toolbarPassed_1bjpgj2"
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="info"
              inverse={false}
              key="toolbar-info"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Info"
              transform={null}
            />
          </div>
        </NavItem>
        <UncontrolledDropdown
          inNavbar={true}
          nav={true}
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <DropdownToggle
              aria-haspopup={true}
              className="toolbar_l6n5rs-o_O-toolbarPassed_1bjpgj2"
              color="secondary"
              nav={true}
            >
              <FontAwesomeIcon
                border={false}
                className="toolbarButton_1d6e8oq"
                fixedWidth={false}
                flip={null}
                icon="filter"
                inverse={false}
                key="toolbar-filter"
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Choose filter"
                transform={null}
              />
            </DropdownToggle>
          </div>
          <DropdownMenu
            className="filterDropdown_biw13d"
            flip={true}
            tag="div"
          >
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={true}
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                  value="all"
                />
                 
                All
              </Label>
            </DropdownItem>
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={false}
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                  value="fail"
                />
                 
                Failed only
              </Label>
            </DropdownItem>
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={false}
                  name="filter"
                  onChange={[Function]}
                  type="radio"
                  value="pass"
                />
                 
                Passed only
              </Label>
            </DropdownItem>
            <DropdownItem
              divider={true}
              tag="button"
              toggle={true}
            />
            <DropdownItem
              className="dropdownItem_1bl0z5g"
              tag="button"
              toggle={false}
            >
              <Label
                check={true}
                className="filterLabel_17zzvy"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <Input
                  checked={false}
                  name="displayEmptyTest"
                  onChange={[Function]}
                  type="checkbox"
                />
                 
                Hide empty testcase
              </Label>
            </DropdownItem>
          </DropdownMenu>
        </UncontrolledDropdown>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="print"
              inverse={false}
              key="toolbar-print"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Print page"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="tags"
              inverse={false}
              key="toolbar-tags"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Display tags"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <div
            className="buttonsBar_6zsntc"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="question-circle"
              inverse={false}
              key="toolbar-question"
              listItem={false}
              mask={null}
              onClick={[Function]}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Help"
              transform={null}
            />
          </div>
        </NavItem>
        <NavItem
          tag="li"
        >
          <a
            className="buttonsBar_6zsntc"
            href="http://testplan.readthedocs.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FontAwesomeIcon
              border={false}
              className="toolbarButton_1d6e8oq"
              fixedWidth={false}
              flip={null}
              icon="book"
              inverse={false}
              key="toolbar-document"
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Documentation"
              transform={null}
            />
          </a>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="HelpModal"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Help
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      This is filter box help!
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="light"
        onClick={[Function]}
        tag="button"
      >
        Close
      </Button>
    </ModalFooter>
  </Modal>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="infoModal"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    size="lg"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Information
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      No information to display.
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="light"
        onClick={[Function]}
        tag="button"
      >
        Close
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
